irohad/ametsuchi/impl/rocksdb_specific_query_executor.cpp

```diff
@@ -41,6 +41,38 @@
 using namespace iroha;
 using namespace iroha::ametsuchi;
 
+namespace {
+  struct PaginationBounds {
+    using HeightType = shared_model::interface::types::HeightType;
+    using TimestampType = shared_model::interface::types::TimestampType;
+
+    HeightType heightFrom;
+    HeightType heightTo;
+
+    TimestampType tsFrom;
+    TimestampType tsTo;
+  };
+
+  template <typename T>
+  inline T getMinValue() {
+    return std::numeric_limits<T>::min();
+  }
+
+  template <typename T>
+  inline T getMaxValue() {
+    return std::numeric_limits<T>::max();
+  }
+
+  template <typename T>
+  inline T getValueOr(std::optional<T> value, T def) {
+    static_assert(std::is_integral_v<T> || std::is_floating_point_v<T>,
+                  "T must be numeric or derived from numeric.");
+    if (value)
+      return *value;
+    return def;
```

Comment:
Why not to use std::optional abilities? https://en.cppreference.com/w/cpp/utility/optional/value_or
```suggestion
   return value.value_or(def);
```

Reply:
+

Reply:
+

Reply:
??